apps/stonks/stonks.star

```diff
@@ -0,0 +1,239 @@
+"""
+Applet: Stonks
+Summary: Fetches stock data from polygon.io
+Description: Shows stock price for given ticker and optionally portfolio value, see https://stonks.market to get started
+Author: harrisonpage + tfarnon
+
+v1.0
+Released 15-Sep-2024.
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+# 30 minute cache
+CACHE_TIME = 1800
+
+# increase value to clear cache
+VERSION = "v2"
+
+DEFAULT_SYMBOL = "AAPL"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "apikey",
+                name = "Polygon.io API Key",
+                desc = "Basic (free) API tier is fine",
+                icon = "moneyBill",
+            ),
+            schema.Text(
+                id = "symbol",
+                name = "Symbol",
+                desc = "e.g. GOOG, APPL, etc.",
+                icon = "moneyBill",
+            ),
+            schema.Text(
+                id = "portfolio",
+                name = "Shares Owned",
+                desc = "# of shares you own",
+                icon = "dollarSign",
+            ),
+            schema.Color(
+                id = "symbol_color",
+                name = "Symbol Color",
+                desc = "Color of ticker symbol",
+                icon = "brush",
+                default = "#00FF00",
+            ),
+            schema.Color(
+                id = "price_color",
+                name = "Price Color",
+                desc = "Color of stock price",
+                icon = "brush",
+                default = "#FFFFFF",
+            ),
+            schema.Color(
+                id = "portfolio_color",
+                name = "Portfolio Color",
+                desc = "Color of portfolio",
+                icon = "brush",
+                default = "#FF0000",
+            ),
+        ],
+    )
+
+def format_currency(amount):
+    """Format currency for US"""
+    str_amount = str(amount).split(".")
+    dollars = str_amount[0]
+    cents = str_amount[1] if len(str_amount) > 1 else "00"
+    cents = (cents + "0")[:2]
+    return "${}.{}".format(dollars, cents)
+
+def get_stock_quote(url, symbol):
+    """Fetch quote from polygon.io or cache"""
+    cache_key = "stonks-{0}-{1}".format(symbol, VERSION)
+    cached_data = cache.get(cache_key)
```

Comment:
You can replace these calls to the cache module by passing `ttl_seconds` directly to `http.get()`.

Reply:
Does that cache across all instances of the app like (I think) this cache does? I could see it nice to cache stock data across all installations since the data is universal.

Reply:
Ah, I see what you're saying. Clever!